journals/apps/api/filters.py

```diff
@@ -50,6 +51,20 @@ def filter_latest(self, queryset, name, value):  # pylint: disable=unused-argume
         filtered_queryset = queryset.filter(uuid__in=filtered_uuids)
         return filtered_queryset
 
+    def filter_xblock_id(self, queryset, name, value):  # pylint: disable=unused-argument
+        """
+        Get the Journal from video block_id and then filter on journal.
+        """
+        if not value:
+            return queryset
+        qs = queryset
+        try:
+            journal_uuid = JournalPage.objects.get(videos=Video.objects.get(block_id=value)).get_journal().uuid
```

Comment:
I think this is going to raise `MultipleObjectsReturned ` if a video is used in multiple journal pages. Could you please check?

Reply:
@ziafazal yes, you are right. It returns `MultipleObjectsReturned`.
`journal_pages = JournalPage.objects.filter(videos=self)`
` journal_uuids = [journal_page.get_journal().uuid for journal_page in journal_pages]`
How should I get a single uuid from video block?